### PR TITLE
Complete Course animation

### PR DESCRIPTION
**Pull request by benceisonline**
## Description

Course completion animation, and small fix to section animation. The matching backend should be merged first.

## Changes

Mainly added files for the course completion and commented extra points out on section completion screen.

## Related Issues

https://github.com/orgs/Educado-App/projects/9/views/1?pane=issue&itemId=44144208

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

## Screenshots (if applicable)

Screenshot 2023-11-28 at 14 35 47
Screenshot 2023-11-28 at 14 35 41
Screenshot 2023-11-28 at 14 35 34
Screenshot 2023-11-28 at 14 32 33

## If mobile/frontend pull request, what version of the backend is it stable, and running on? 

Branch: gamified_dev_copy_nov28

Commit id: 579d445

## Notes for Reviewers

Some comments are relevant for future developers. Hack database to display course completion:
- Go into the student doc, go to the finance course in the field courses, and set the first and second elements in the array of sections to be isCompleted = true

### CODE REVIEW

File: eml/api/userApi.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ export const registerUser = async (obj) => {
 export const loginUser = async (obj) => {
   try {
     const res = await client.post('/api/auth/login', obj);
-    console.log('User successfully registered');
+    console.log('User successfully logged in');
```
Comment (Vagn99) [inline eml/api/userApi.js:60]: Begone log!

File: eml/components/courses/completeCourse/Certification.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+          }}
+          source={require('../../../assets/animations/diploma.json')}
+          autoPlay
+        />
+      </View>
+      
+      <View className="-mt-5">
+        <TouchableOpacity 
+          // TODO: for the cert group
+          onPress={() => { console.log('Navigate to certification'); }}
```
Comment (Vagn99) [inline eml/components/courses/completeCourse/Certification.jsx:30]: Begone log!

**Review by Vagn99 — CHANGES_REQUESTED** [review]
Coppa logs

**Review by carlryskov — APPROVED** [review]
Diplom og image forbliver til næste år? Eller er det meningen der er hardcodede? 

Derudover ser det godt ud

File: eml/services/utilityFunctions.js [hunk 3]
```diff
@@ -252,22 +251,27 @@ export function findIndexOfUncompletedComp(student, courseId, sectionId) {
 }
 
 export async function handleLastComponent(comp, course, navigation) {
-  const student = await StorageService.getStudentInfo();
-  const isComplete = isSectionCompleted(student, comp.parentSection);
+
+  // For future reference
+  // const student = await StorageService.getStudentInfo();
+  // const isComplete = isSectionCompleted(student, comp.parentSection);
```
Comment (sofiiagran) [inline eml/services/utilityFunctions.js:257]: out of curiosity, what does this mean? :))

File: eml/services/utilityFunctions.js [hunk 4]
```diff
@@ -252,22 +251,27 @@ export function findIndexOfUncompletedComp(student, courseId, sectionId) {
...
+      },
+    ],
+  });
+
+  // For future reference
+  // if (isComplete) { 
+  // Code above with naviagtion
+  // } else {
+  //   console.log('Section not complete - navigate to retry exercises TBD');
+  // }
```
Comment (sofiiagran) [inline eml/services/utilityFunctions.js:276]: same, what does this mean/how can it be used?

File: eml/components/courses/completeCourse/Certification.jsx (hunk 2 above)
Comment (sofiiagran) [reply eml/components/courses/completeCourse/Certification.jsx]: enig

**Review by sofiiagran — APPROVED** [review]
I approve, but can you make PR backend from dev to staging, so that it can be used with staging backend as well?

File: eml/components/courses/completeCourse/Certification.jsx (hunk 2 above)
Comment (benceisonline) [reply eml/components/courses/completeCourse/Certification.jsx]: log gone 💥💥💥

File: eml/api/userApi.js (hunk 1 above)
Comment (benceisonline) [reply eml/api/userApi.js]: log gone 💥💥💥

File: eml/services/utilityFunctions.js (hunk 3 above)
Comment (benceisonline) [reply eml/services/utilityFunctions.js:257]: the future reference stuff is left there for next year's students who will get a PBI for retrying exercises.

File: eml/services/utilityFunctions.js (hunk 4 above)
Comment (benceisonline) [reply eml/services/utilityFunctions.js:276]: see other answer ≽ܫ≼

**Review by Vagn99 — APPROVED** [review]

**Pull request merged**